src/library/scala/collection/IterableOnce.scala

```diff
@@ -912,7 +912,7 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
     *   @tparam  B    the result type of the `+` operator.
     *   @return       the sum of all elements of this $coll with respect to the `+` operator in `num`.
     */
-  def sum[B >: A](implicit num: Numeric[B]): B = foldLeft(num.zero)(num.plus)
+  def sum[B >: A](implicit num: Numeric[B]): B = reduceOption(num.plus).getOrElse(num.zero)
```

Comment:
Not sure if the `Option` of `reduceOption` would be much of a cost, to make it worth it to just use
```Scala
if (isEmpty) num.zero else reduce(num.plus)
```
(this is pseudo-code I am not sure of exact methods).